libs/platform/user/ebpf_platform_user.cpp

```diff
@@ -1408,3 +1413,37 @@ ebpf_utf8_string_to_unicode(_In_ const ebpf_utf8_string_t* input, _Outptr_ wchar
     ebpf_free(unicode_string);
     return retval;
 }
+
+_Ret_maybenull_ ebpf_process_state_t*
+ebpf_allocate_process_state()
```

Comment:
Can you check Fault injection needed for all the new functions introduced?

I believe it is needed.

Reply:
This PR is adding the following APIs:

```
ebpf_allocate_process_state
ebpf_platform_reference_process
ebpf_platform_dereference_process
ebpf_platform_attach_process
ebpf_platform_detach_process
```
Only one of these APIs is expected to fail -- `ebpf_allocate_process_state`. And since that API calls `ebpf_allocate()`, it is already covered for fault injection.

Reply:
Thanks for checking it. Please add a comment for ebpf_allocate_process_state with
Fault injection as skipped.

Reply:
Added.